src/react/utils.js

```diff
@@ -566,6 +566,46 @@ export function evaluateWithNestedEffects(
   }
 }
 
+// create a new effects object with the bindings, property bindings
+// and created objects of both effects combined into a single effects
+// also ensure use the value and generator from the target effect are
+// used in the new effects object
+function createMergedEffects(realm: Realm, target: Effects, source: Effects): Effects {
```

Comment:
It seems pretty leaky to have this in the React utils. I've change the structure of this in the past and probably will again so would be nice to have this centralized... if this is a legit technique.

It should also be named something else because for conflicts it'll prefer one effect over another. Effects can't just be arbitrarily joined since they have conflicts.